pkg/aws/client.go

```diff
@@ -1213,62 +1212,40 @@ func (c *awsClient) DeleteSecretInSecretsManager(secretArn string) error {
 	return nil
 }
 
-func (c *awsClient) GetSecurityGroupIds(vpcId string) ([]*ec2.SecurityGroup, error) {
+func (c *awsClient) GetSecurityGroupIds(vpcId string) ([]*ec2types.SecurityGroup, error) {
 	describeSecurityGroupsInput := &ec2.DescribeSecurityGroupsInput{
-		Filters: []*ec2.Filter{
+		Filters: []ec2types.Filter{
 			{
 				Name:   aws.String("vpc-id"),
-				Values: aws.StringSlice([]string{vpcId}),
+				Values: []string{vpcId},
 			},
 		},
 	}
-	securityGroups := []*ec2.SecurityGroup{}
-	err := c.ec2Client.DescribeSecurityGroupsPages(describeSecurityGroupsInput,
-		func(page *ec2.DescribeSecurityGroupsOutput, lastPage bool) bool {
-			for _, sg := range page.SecurityGroups {
-				if tags.Ec2ResourceHasTag(sg.Tags, tags.RedHatManaged, strconv.FormatBool(true)) {
-					continue
-				}
-				if aws.StringValue(sg.GroupName) == "default" {
-					continue
-				}
-				securityGroups = append(securityGroups, sg)
-			}
-			return page.NextToken != nil
-		})
+	securityGroups := []*ec2types.SecurityGroup{}
+	resp, err := c.ec2Client.DescribeSecurityGroups(context.Background(), describeSecurityGroupsInput)
 	if err != nil {
-		return []*ec2.SecurityGroup{}, err
+		return []*ec2types.SecurityGroup{}, err
 	}
-	return securityGroups, nil
-}
 
-// CustomRetryer wraps the aws SDK's built in DefaultRetryer allowing for
-// additional custom features
-type CustomRetryer struct {
-	client.DefaultRetryer
-}
-
-// ShouldRetry overrides the SDK's built in DefaultRetryer adding customization
-// to not retry 5xx status codes.
-func (r CustomRetryer) ShouldRetry(req *request.Request) bool {
-	if req.HTTPResponse.StatusCode >= 500 {
-		return false
-	}
-	logger := logging.NewLogger()
-	if strings.Contains(req.Error.Error(), "Throttling") {
-		logger.Warn("Throttling Rate limit exceeded. Retrying the request again")
+	for _, secGroup := range resp.SecurityGroups {
+		if c.Ec2ResourceHasTag(secGroup.Tags, tags.RedHatManaged, strconv.FormatBool(true)) {
+			continue
+		}
+		if aws.ToString(secGroup.GroupName) == "default" {
+			continue
+		}
+		sg := secGroup
+		securityGroups = append(securityGroups, &sg)
```

Comment:
It seems like `DescribeSecurityGroups` returns a slice of `[]SecurityGroup`, do we still need the casting to pointers here? or can we change the signature of the function to return a slice of `SecurityGroup`?